gammapy/makers/utils.py

```diff
@@ -35,23 +35,40 @@ def make_map_exposure_true_energy(pointing, livetime, aeff, geom):
         Effective area
     geom : `~gammapy.maps.WcsGeom`
         Map geometry (must have an energy axis)
+    use_region_center: bool
+        If geom is a RegionGeom, whether to just
+        consider the values at the region center
+        or the insted the average over the whole region
 
     Returns
     -------
     map : `~gammapy.maps.WcsNDMap`
         Exposure map
     """
-    offset = geom.separation(pointing)
-    energy_true = geom.axes["energy_true"].center
+    if not use_region_center:
+        wcs_geom = geom.to_wcs_geom()
```

Comment:
Here is one last suggestion to make it "perfect" in terms of performance. Creating a full WCS geometry here results in using a full 2d array of coordinates to be used for evaluation in `aeff.evaluate()`, in practice many of pixels of the 2d array are empty anyway (the ones outside the region). So the proposal here is to only use the flat coordinate list returned by `RegionGeom.get_wcs_coord()`. Look at the following minimal example:
```python3
import numpy as np

energy = np.array([1, 3, 10, 30])

a = np.ones((3, 2))
b = np.ones((6))
b_weights = 0.5 * np.ones((6,))

# a is a 2d array of coordinates
data_wcs = a * energy[:, np.newaxis, np.newaxis]
print(data_wcs.shape)

# b is a 1d array of coordinates
values = (b * energy[:, np.newaxis, np.newaxis])

# divide by sum of weight for the mean
data_region = np.sum(values * b_weights, axis=-1, keepdims=True) / b_weights.sum() 
print(data_region.shape) 
```
I now realise I was maybe a bit fast with the introduction of the `.get_wcs_weights()` method. Maybe we should modify `.get_wcs_coords()` to return both: the valid coordinates and the corresponding weights for each coordinate to represent the region. So the typical usage would be:
```python3
coords, weights = region_geom.get_wcs_coords()

values = np.sum(something.evaluate(coords) * weights)
```
Does this make sense?

Reply:
that is a good idea. I agree that it is better. Indeed the "unnecessary" pixels were bothering me too and I think the what you propose of merging `get_wcs_coords` and `get_wcs_weights` is nice too. I'll implement it tomorrow.